public/app/features/explore/QueryLibrary/QueryTemplatesTable/ActionsCell.tsx

```diff
@@ -84,15 +83,15 @@ function ActionsCell({ queryTemplate, rootDatasourceUid, queryUid }: ActionsCell
           queryLibraryTrackAddOrEditDescription();
         }}
       />
-      <ExploreRunQueryButton
-        queries={queryTemplate.query ? [queryTemplate.query] : []}
-        rootDatasourceUid={rootDatasourceUid}
-        variant="primary"
-        onClick={() => {
-          setDrawerOpened(false);
-          queryLibraryTrackRunQuery(queryTemplate.datasourceType || '');
-        }}
-      />
+      {QueryActionButton && (
```

Comment:
Action button is passed down instead of being hardcoded to Explore specific handler